src/ynEIGEN/WithdrawalsProcessor.sol

```diff
@@ -0,0 +1,432 @@
+
+    // assets
+    IERC20 private constant STETH = IERC20(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+    IwstETH private constant WSTETH = IwstETH(0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0);
+    IERC4626 private constant WOETH = IERC4626(0xDcEe70654261AF21C44c093C300eD3Bb97b78192);
+    IERC20 private constant OETH = IERC20(0x856c4Efb76C1D1AE02e20CEB03A2A6a08b0b8dC3);
+
+    // used to prevent rounding errors
+    uint256 private constant MIN_DELTA = 1000;
+
+    // roles
+    bytes32 public constant KEEPER_ROLE = keccak256("KEEPER_ROLE");
+
+    //
+    // Constructor
+    //
+    constructor(
+        address _withdrawalQueueManager,
+        address _tokenStakingNodesManager,
+        address _assetRegistry,
+        address _ynStrategyManager,
+        address _delegationManager,
+        address _yneigen,
+        address _redemptionAssetsVault,
+        address _wrapper
+    ) {
+        if (
+            _withdrawalQueueManager == address(0) || _tokenStakingNodesManager == address(0)
+                || _assetRegistry == address(0) || _ynStrategyManager == address(0) || _delegationManager == address(0)
+                || _yneigen == address(0) || _redemptionAssetsVault == address(0) || _wrapper == address(0)
+        ) revert InvalidInput();
+
+        withdrawalQueueManager = IWithdrawalQueueManager(_withdrawalQueueManager);
+        tokenStakingNodesManager = ITokenStakingNodesManager(_tokenStakingNodesManager);
+        assetRegistry = IAssetRegistry(_assetRegistry);
+        ynStrategyManager = IYieldNestStrategyManager(_ynStrategyManager);
+        delegationManager = IDelegationManager(_delegationManager);
+        yneigen = IynEigen(_yneigen);
+        redemptionAssetsVault = IRedemptionAssetsVault(_redemptionAssetsVault);
+        wrapper = IWrapper(_wrapper);
+    }
+
+    function initialize(address _owner, address _keeper) public initializer {
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _owner);
+        _grantRole(KEEPER_ROLE, _keeper);
+
+        minPendingWithdrawalRequestAmount = 0.1 ether;
+    }
+
+    //
+    // view functions
+    //
+
+    /// @notice IDs of the queued, completed, and processed withdrawals, used for internal accounting
+    /// @return The IDs
+    function ids() external view returns (IDs memory) {
+        return _ids;
+    }
+
+    /// @notice Gets the queued withdrawal at the given ID
+    /// @param _id The ID of the queued withdrawal
+    /// @return The queued withdrawal
+    function queuedWithdrawals(
+        uint256 _id
+    ) external view returns (QueuedWithdrawal memory) {
+        return _queuedWithdrawals[_id];
+    }
+
+    /// @notice Checks if withdrawals should be queued
+    /// @return True if withdrawals should be queued, false otherwise
+    function shouldQueueWithdrawals() external view returns (bool) {
+        return withdrawalQueueManager.pendingRequestedRedemptionAmount() - totalQueuedWithdrawals
```

Comment:
since nothing is queued now, this starting at 0 looks good!

totalQueuedWithdrawals